File: shoring/compat/src/main/java/com/panopset/compat/Randomop.java
package com.panopset.compat;

import java.security.SecureRandom;

public enum Randomop {

  INSTANCE;

  private final SecureRandom secureRandom = new SecureRandom();

  public static int random(int min, int max) {
    if (min == max) {
      return min;
    } else {
      int shift = min;
      if (min > max) {
        shift = max;
      }
      return INSTANCE.secureRandom.nextInt(Math.abs(max - min) + 1) + shift;
    }
  }

  public static void nextBytes(byte[] bytes) {
    INSTANCE.secureRandom.nextBytes(bytes);
  }

  public static int nextInt() {
    return INSTANCE.secureRandom.nextInt();
  }

  public static long nextLong() {
    return INSTANCE.secureRandom.nextLong();
  }
}

File: code/shoring/compat/src/main/kotlin/com/panopset/compat/Randomop.kt
package com.panopset.compat

import java.io.StringWriter
import java.security.SecureRandom
import kotlin.math.abs

object Randomop {

    private val secureRandom = SecureRandom()

    fun random(min: Int, max: Int): Int {
        return if (min == max) {
            min
        } else {
            var shift = min
            if (min > max) {
                shift = max
            }
            secureRandom.nextInt((abs((max - min).toDouble()) + 1).toInt()) + shift
        }
    }

    fun nextBytes(bytes: ByteArray?) {
        secureRandom.nextBytes(bytes)
    }

    fun nextInt(): Int {
        return secureRandom.nextInt()
    }

    fun nextLong(): Long {
        return secureRandom.nextLong()
    }

    fun getRandomHexString(length: Int): String {
        val sw = StringWriter()
        for (i in 0 until length) {
            sw.append(String.format("%s", random(0, 15)))
        }
        return sw.toString()
    }
}
